### DISCUSSION TITLE
Stochastic Tools module, POD-RB for generalized eigenvalue problems?

### DISCUSSION THREAD

**Discussion by crb3874** [idaholab/moose#19645] [Q&A Modules: General] [answered]
I'm looking into using the Stochastic Tools module for generating surrogate POD-RB models for neutron diffusion problems similar to the example [here](https://mooseframework.inl.gov/modules/stochastic_tools/examples/pod_rb_surrogate.html). However, I see that the listed requirement is for the problem to be a scalar-valued linear steady-state PDE with a suitable affine decomposition, meaning a problem like A(mu) u = b(mu). I'd like to do this as an eigenvalue problem with fission, so a problem of the form A(mu) u = lambda B(mu) u - examples like this using POD/PGD have appeared in the literature, and I've implemented it outside of MOOSE. 

I understand this is not currently available functionality - before I go too deep down the rabbit hole of trying to make modifications on my branch (and likely getting it wrong right away)... is this feasible with the current implementation/is there any advice on where to start looking to make changes/additions? Thanks in advance!

**Comment by GiudGiud** [comment, marked as answer]
@zachmprince

**Reply by GiudGiud** [reply]
@somu15 @zachmprince could you please share your comments on this?

**Reply by crb3874** [reply]
@GiudGiud Thank you for doubling back on this - I think this question can be resolved though, as I was able to get this working for myself without too much issue (turned out to be much simpler than expected).
